custom-targets/vertex-ai/model-deployer/render.go

```diff
@@ -0,0 +1,226 @@
+	}
+	fmt.Printf("Downloaded render input archive from %s\n", inURI)
+
+	out, err := r.renderDeployModelRequest()
+	if err != nil {
+		return nil, fmt.Errorf("error rendering deploy model params: %v", err)
+	}
+
+	fmt.Printf("Uploading deployed model manifest.\n")
+
+	mURI, err := r.req.UploadArtifact(ctx, r.gcsClient, "manifest.yaml", &clouddeploy.GCSUploadContent{Data: out})
+	if err != nil {
+		return nil, fmt.Errorf("error uploading deployed model manifest: %v", err)
+	}
+
+	fmt.Printf("Uploaded deployed model manifest to %s\n", mURI)
+
+	return &clouddeploy.RenderResult{
+		ResultStatus: clouddeploy.RenderSucceeded,
+		ManifestFile: mURI,
+	}, nil
+}
+
+// renderDeployModelRequest generates a DeployModelRequest object and returns its definition as a yaml-formatted string
+func (r *renderer) renderDeployModelRequest() ([]byte, error) {
+
+	if err := applyDeployParams(r.params.configPath); err != nil {
+		return nil, fmt.Errorf("cannot apply deploy parameters to configuration file: %v", err)
+	}
+
+	// blank deployed model template
+	deployedModel := &aiplatform.GoogleCloudAiplatformV1DeployedModel{}
+
+	configuration, err := loadConfigurationFile(r.params.configPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to obtain configuration data: %v", err)
+	}
+
+	if err = yaml.Unmarshal(configuration, deployedModel); err != nil {
+		return nil, fmt.Errorf("unable to parse configuration data into DeployModel object: %v", err)
+	}
+
+	model, err := fetchModel(r.aiPlatformService, r.params.model)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch model: %v", err)
+	}
+
+	modelNameWithVersionId := resolveModelWithVersion(model)
+	if err != nil {
+		return nil, fmt.Errorf("unable to resolve model version: %v", err)
+	}
+
+	if err := validateRequest(modelNameWithVersionId, r.params.endpoint, r.params.minReplicaCount, deployedModel); err != nil {
+		return nil, fmt.Errorf("manifest validation failed: %v", err)
+	}
+	deployedModel.Model = modelNameWithVersionId
+
+	if deployedModel.DedicatedResources == nil {
+		deployedModel.DedicatedResources = &aiplatform.GoogleCloudAiplatformV1DedicatedResources{MinReplicaCount: r.params.minReplicaCount}
+	}
+
+	if deployedModel.DedicatedResources.MinReplicaCount == 0 {
+		deployedModel.DedicatedResources.MinReplicaCount = r.params.minReplicaCount
+	}
+
+	// deploy model params requires this field to be non-nil. Setting to the default "n1-standard-2"
+	// if it's not already set
+	if deployedModel.DedicatedResources.MachineSpec == nil {
+		deployedModel.DedicatedResources.MachineSpec = &aiplatform.GoogleCloudAiplatformV1MachineSpec{MachineType: "n1-standard-2"}
+	}
+
+	if deployedModel.DedicatedResources.MachineSpec.MachineType == "" {
+		deployedModel.DedicatedResources.MachineSpec.MachineType = "n1-standard-2"
+	}
+
+	percentage := int64(r.req.Percentage)
+	trafficSplit := map[string]int64{}
+	trafficSplit["0"] = percentage
```

Comment:
what's the "0" key represent here? Does this deserve a comment explaining it?

Reply:
added comment. Its a stand in for the current model.